### ISSUE TITLE
Bug with g_sounds buffer

### ISSUE THREAD

**Issue by JosensGames** [opentibiabr/otclient#399]
### Priority

Low

### Area

- [ ] Data
- [X] Source
- [ ] Docker
- [ ] Other

### What happened?

Hello, I'm integrating sounds on my client and I found a bug with the client audio buffer;

When I play a lot of sounds the buffer breaks, for example, I you type this one in the terminal
```
for i = 1, 300 do
    scheduleEvent(function() g_sounds.play(X) end, i*20)
end
```

The client will begin to print this warn:

`ERROR: unable to refill audio buffer for '/client_audio/files/sfx/sfx_combat_arrowshoot_2.ogg': Out of Memory`.

I think, the problem is when a sounds being played, the client audio buffer begins to be filled up, and boom, it explodes after playing a certain amount of sounds.

### What OS are you seeing the problem on?

Linux, Windows, MacOS

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by mehah** [comment]
I'm trying to understand why you want to play trillions of sounds at once.

**Comment by mehah** [comment]
I'm trying to understand why you want to play many sounds at the same time.

```lua
for i = 1, 300 do
    scheduleEvent(function() g_sounds.play(X) end, i*20)
end
```

20 = 20ms
2000 = 2seg

if the time between sounds is reasonable, there will be no memory overflow.

**Comment by JosensGames** [comment]
I know, because it's an example.

In the real game, for example if two players begin a battle, assuming that the client contains sounds, it also crashes...
Of course, in the case of two creatures in a battle, the client crash will delay because the amount of sounds in the screen are less.

Is not the amount per seconds, is after playing certain amount of sounds.

**Comment by mehah** [comment]
@JosensGames 
I already fixed the crash problem, just look at the last commits.

https://github.com/mehah/otclient/commit/601f89d78f0439739385d5de13f92d36e975ead2

**Comment by mehah** [comment]
no user feedback.

**Comment by JosensGames** [comment]
Hello @mehah, after a year-long break, I decided to return to my server, which I had abandoned due to the sound bug.

What I was trying to explain earlier is that every time you play a sound, whether in a channel or not, the sound fills up the program's buffer memory until the client crashes.

In this case, I played an .ogg file and observed how the program's memory becomes saturated. Approximately, when it reaches 400MB and beyond, it will throw the following error when trying to play any sound:

**ERROR: unable to refill audio buffer for '/data/sounds/sfx/sfx_skill.ogg': Out of Memory**


https://github.com/mehah/otclient/assets/9597129/32fc8675-b38c-41fd-aef5-eeb1b958b425

It is worth noting that I tried a lot of ways through Lua to clear the client's audio buffer, but there was no result, neither with **g_sounds.stopAll()** nor anything else.

Should we implement a new function to clear the memory of a sound once it stops playing? Or perhaps a similar function in the source code?
